src/main/java/duke/file/Storage.java

```diff
@@ -0,0 +1,142 @@
+                        tasks.add(new Deadline(DEADLINE, parsedStoredTask[2], true, parsedStoredTask[3]));
+                    } else {
+                        tasks.add(new Deadline(DEADLINE, parsedStoredTask[2], false, parsedStoredTask[3]));
+                    }
+                    break;
+                default:
+                    break;
+                }
+            }
+        } catch (FileNotFoundException e) {
+            Ui.showFileError(MISSING_FILE);
+        }
+    }
+
+
+    /**
+     * called when file does not exist and creates a new file
+     *
+     * @return file when new file is created and null if no file is created.
+     */
+    public static File createFile() {
+        try {
+            File file = new File("data/dukefile.txt");
+            file.createNewFile();
+            return file;
+        } catch (Exception e) {
+            Ui.showFileError(FAILED_CREATION);
+        }
+        return null;
+    }
+
+    /**
+     * called to check on the existance of directory and file. Function creates missing directory and files when necessary.
+     *
+     * @return file after ensuring making sure directory and file is set.
+     */
+    public static File manageFile() {
+
+        File directory = new File("data");
+        File file = new File("data/dukefile.txt");
+        if (!directory.exists()) {
+            directory.mkdir();
+            System.out.println("Creating new directory");
+        }
+        if (!file.exists()) {
+            file = createFile();
+            System.out.println("* Created new file for use *");
+        } else {
+            System.out.println("* Existing file is loaded and ready to be used *");
+        }
+        return file;
+    }
+
+    /**
+     * updates dukefile.txt when called for each valid user command which requires updating of the tasks array.
+     *
+     * @param tasks the array containing the entries which would be converted to text format.
+     */
+    public static void writeToFile(ArrayList<Task> tasks) {
+        try {
+            FileWriter file = new FileWriter("data/dukefile.txt");
+            String textToWrite = "";
+            for (Task task : tasks) {
+                switch (task.getTaskType()) {
+                case DEADLINE:
+                    textToWrite = String.format(task.getTaskType() + " | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + task.getDateTime());
+
+                    break;
+                case EVENT:
+                    textToWrite = String.format(task.getTaskType() + " | " + task.getStatusIcon() + " | " + task.getDescription() + " | " + task.getDateTime());
+
+                    break;
+                case TODO:
+                    textToWrite = String.format(task.getTaskType() + " | " + task.getStatusIcon() + " | " + task.getDescription());
+                    break;
+                default:
+                    break;
+                }
```

Comment:
Each task type should have a `toString` (or `toFileString` or similar) method that gets called here. Basically, each task should know how to print itself. It shouldn't be `writeToFile`'s responsibility to know how to format a task. Moreover, this approach isn't very extensible: if you add a new task type, you would have to change this method, which is strange.